mobile/src/navigation/root/root.navigation.tsx

```diff
@@ -16,11 +19,34 @@ const screenOptions: NativeStackNavigationOptions = {
 };
 
 const Root: FC = () => {
+  const { user } = useAppSelector((state) => ({ user: state.auth.user }));
+  const dispatch = useAppDispatch();
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    dispatch(loadCurrentUser()).then(() => {
+      setIsLoading(false);
+    });
+  }, [dispatch, setIsLoading]);
+
+  if (!user && isLoading) {
+    return <Spinner />;
+  }
```

Comment:
Sometimes, please take a look to the frontend part for same logic.In this case, to component/app/app